Sources/Utilities/KMZendeskChatHandler.swift

```diff
@@ -0,0 +1,238 @@
+//
+//  KMZendeskChatHandler.swift
+//  KommunicateChatUI-iOS-SDK
+//
+//  Created by sathyan elangovan on 28/11/22.
+//
+
+import Foundation
+import ChatSDK
+import ChatProvidersSDK
+import KommunicateCore_iOS_SDK
+
+struct KommunicateURL {
+    static let attachmentURL = "https://chat.kommunicate.io/rest/ws/attachment/"
+    static let dashboardURL = "https://dashboard.kommunicate.io/conversations/"
+}
+
+public class KMZendeskChatHandler {
+    
+   public static let shared = KMZendeskChatHandler()
+    var groupId: String = ""
+    var connectionStatus = false
+    var messages:[ALMessage]? = nil
+    var zenChatIntialized = false
+    var userId = ""
+    var isChatTranscriptSent = false
+    
+    public func initiateZendesk(key: String, conversationId: String) {
+        Chat.initialize(accountKey: key)
+        groupId = conversationId
+        zenChatIntialized = true
+        fetchUserId()
+        authenticateUser()
+
+        DispatchQueue.main.asyncAfter(deadline: .now() + 2.0) {
+            self.sendChatInfo()
+            self.sendChatTranscript()
+            
+            let connectionToken = Chat.connectionProvider?.observeConnectionStatus { [self] (connection) in
+                // Handle connection status changes
+                guard connection.isConnected && connectionStatus != true else {
+                   self.connectToZendeskSocket()
+                   return
+                }
+                connectionStatus = true
+                observeChatLogs()
+                if !isChatTranscriptSent {
+                    sendChatTranscript()
+                }
+               
+           }
+           
+       }
+    }
+    
+    func fetchUserId() {
+        guard let id = ALUserDefaultsHandler.getUserId() else {return}
+        self.userId = id
+    }
+    
+    public func getGroupId() -> String {
+        return groupId
+    }
+    func authenticateUser() {
+        let chatAPIConfiguration = ChatAPIConfiguration()
+        chatAPIConfiguration.visitorInfo = VisitorInfo(name: ALUserDefaultsHandler.getDisplayName() ?? "", email: ALUserDefaultsHandler.getEmailId() ?? "", phoneNumber: "")
+        Chat.instance?.configuration = chatAPIConfiguration
+        connectToZendeskSocket()
+    }
+    var chatLogToken : ObservationToken?
+    
+    func observeChatLogs() {
+        guard Chat.connectionProvider?.status == .connected else {
+            connectToZendeskSocket()
+            return
+        }
+        
+        let stateToken = Chat.chatProvider?.observeChatState { (state) in
+            // Handle logs, agent events, queue position changes and other events
```

Comment:
If we are not doing anything, then pass an empty closure.

Reply:
No we have to observe events..that will be a separate ticket.thatswhy I kept it like this